sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/CodeFixVerifier.cs

```diff
@@ -85,9 +85,12 @@ internal static class CodeFixVerifier
                 var codeFixExecuted = false;
                 for (var diagnosticIndexToFix = 0; !codeFixExecuted && diagnosticIndexToFix < diagnostics.Count; diagnosticIndexToFix++)
                 {
-                    var codeActionsForDiagnostic = GetCodeActionsForDiagnostic(codeFixProvider, currentDocument, diagnostics[diagnosticIndexToFix]);
+                    var diagnostic = diagnostics[diagnosticIndexToFix];
+                    var codeActionsForDiagnostic = codeFixProvider.FixableDiagnosticIds.Contains(diagnostic.Id)
```

Comment:
Another option is to do the check before retrieving actions and continue the for directly if the diagnostic is not fixable:

```
if (!codeFixProvider.FixableDiagnosticIds.Contains(diagnostic.Id))
{
    continue;
}   
```